doc/src/devdocs/aot.md

```diff
@@ -0,0 +1,76 @@
+# Ahead of Time Compilation
+
+This document describes the design and structure of the ahead-of-time (AOT) compilation system in Julia. This system is used when generating system images and package images. Much of the implementation described here is located in `aotcompile.cpp`, `staticdata.c`, and `processor.cpp`
+
+## Introduction
+
+Though Julia normally compiles code just-in-time (JIT), it is possible to compile code ahead of time and save the resulting code to a file. This can be useful for a number of reasons:
+1. To reduce the time it takes to start a Julia process.
+2. To reduce the time spent in the JIT compiler instead of executing code (time to first execution, TTFX).
+3. To reduce the amount of memory used by the JIT compiler.
+
+## High-Level Overview
+
+Due to Julia's dynamic nature, AOT compilation is more involved than in other languages.
```

Comment:
```suggestion
```

This doesn't really make sense to me as a comment since it is mostly normative and not informative or descriptive. I personally think Julia is much simpler, but that the baseline "other language" (*cough*C++*cough*) has induced some serious PTSD on the world about how AOT is "supposed" to work.

I would suggest instead saying something about the purpose of this.
```md
The following descriptions are a snapshot of the current implementation
details of the end-to-end pipeline that happens internally when the
user compiles a new AOT module, such as occurs when they type `using Foo`.
These details are likely to change over time as we implement better
ways to handle them, so current implementations may not exactly match
the dataflow and functions described below.
```